arm/Microsoft.Authorization/locks/deploy.bicep

```diff
@@ -0,0 +1,55 @@
+targetScope = 'subscription'
+
+@allowed([
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Required. Set lock level.')
+param level string
+
+@description('Optional. The decription attached to the lock.')
+param notes string = level == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+
+@description('Optional. Enable telemetry via the Customer Usage Attribution ID (GUID).')
+param enableDefaultTelemetry bool = true
+
+@description('Optional. Name of the Resource Group to assign the lock to. If Resource Group name is provided, and Subscription ID is provided, the module deploys at resource group level, therefore assigns the provided lock to the resource group.')
+param resourceGroupName string = ''
+
+@description('Optional. Subscription ID of the subscription to assign the lock to.')
+param subscriptionId string = ''
+
+@sys.description('Optional. Location for all resources.')
+param location string = deployment().location
+
+var enableReferencedModulesTelemetry = false
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name, location)}'
+  location: location
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+module lock_rg 'resourceGroup/deploy.bicep' = if (!empty(resourceGroupName) && !empty(subscriptionId)) {
```

Comment:
I may be mistaken - but shouldn't there be 2 cases? 
- Lock for RG and
- Lock for Subscription

Reply:
Correct, I didn't add it since I couldn't find a way to test it. 
The cross-reference cannot be tested since we don't have a subscription module. 
For testing the subscription case within the lock module we could use our testing subscription, but I'd avoid it since we'd apply a canNotDelete lock on the same subscription in which potentially all the resources are tested (and removed) meanwhile.

Reply:
Yes and no - we should be able to test it manually, which should be good enough - at least to have to code in the module there. The parameter file to run on a regular basis would then obviously not contain that test. A shame it is not possible to avoid the inhertiance :D

Reply:
Not sure I like to provide code which is not **continuously** tested. That's what I meant.

Reply:
And we definitely should, once we have the capabilities available. Modules with features that cannot be continously tested in our environment (e.g. all those we don't have the dependencies for) should be avoided if we can - or in other words: we should try to do our best to actually test them. 

That said, if that's not possible we can only suggest to test manually - or fundamentally throw the correponding code out. So far we accepted code that we knew was working at a point in time instead of outright rejecting it because our CI environment is limited.